apps/dashboard/app/controllers/concerns/batch_connect_concern.rb

```diff
@@ -22,4 +22,16 @@ def bc_dev_app_groups
       apps: nav_dev_apps.select(&:batch_connect_app?)
     )
   end
+
+  def bc_custom_apps_group
+    if !@user_configuration.apps_menu.blank?
+      # Apps menu override takes precedence
+      NavBar.items([@user_configuration.apps_menu]).first
```

Comment:
Casting to an array then choosing first seems odd. Shouldn't `apps_menu` have the same interface as `nav_bar` - i.e., an array of items (groups I guess)?